src/CoreWCF.WebHttp/src/CoreWCF/Channels/JsonMessageEncoderFactory.cs

```diff
@@ -0,0 +1,621 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using CoreWCF.Diagnostics;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Channels
+{
+    internal class JsonMessageEncoderFactory : MessageEncoderFactory
+    {
+        private static readonly ContentEncoding[] _applicationJsonContentEncoding = GetContentEncodingMap(JsonGlobals.ApplicationJsonMediaType);
+        private JsonMessageEncoder _messageEncoder;
+
+        public JsonMessageEncoderFactory(Encoding writeEncoding, int maxReadPoolSize, int maxWritePoolSize, XmlDictionaryReaderQuotas quotas, bool crossDomainScriptAccessEnabled)
+        {
+            _messageEncoder = new JsonMessageEncoder(writeEncoding, maxReadPoolSize, maxWritePoolSize, quotas, crossDomainScriptAccessEnabled);
+        }
+
+        public override MessageEncoder Encoder
+        {
+            get { return _messageEncoder; }
+        }
+
+        public override MessageVersion MessageVersion
+        {
+            get { return _messageEncoder.MessageVersion; }
+        }
+
+        internal static string GetContentType(WebMessageEncodingBindingElement encodingElement)
+        {
+            if (encodingElement == null)
+            {
+                return WebMessageEncoderFactory.GetContentType(JsonGlobals.ApplicationJsonMediaType, TextEncoderDefaults.Encoding);
+            }
+            else
+            {
+                return WebMessageEncoderFactory.GetContentType(JsonGlobals.ApplicationJsonMediaType, encodingElement.WriteEncoding);
+            }
+        }
+
+        static ContentEncoding[] GetContentEncodingMap(string mediaType)
+        {
```

Comment:
Accessibility modifier

Reply:
This is addressed in fa64f8fd703f0284110f2f879ed6896bf89d8ac6.